package.json

```diff
@@ -0,0 +1,6 @@
+{
+  "devDependencies": {
+    "prettier": "^2.8.0",
+    "prettier-plugin-solidity": "1.0.0-beta.19"
+  }
+}
```

Comment:
I have a few questions regarding this:

- does this only work if `package.json` is on the top level?
- how will this interact with `package.json` from the frontend?
- if I say `npm install`, that will store all packages under `node_modules`, which means there would possibly be duplicates between the frontend `node_modules` and toplevel `node_modules`, right?

Reply:
- Yes, `package.json` needs to be in the root for this to work.
- There is no interaction between these 2 folders. Each folder that has a separate `package.json` file is considered its own "project" by npm
- Yes, there can be duplicates in the separate `node_modules` folder but that is not a big problem. 
A little explainer on how npm looks for the `package.json` file when you run `npm install`: 
When you run `npm install` npm goes bottom-to-top trying to find the first `package.json` it can use for the installation process. So if you are inside the `frontend` folder npm will use the `package.json` that is found inside that folder. when you are in `contracts` folder for example, `npm install` will use the `package.json` that is located in the top-most folder (the root). Same goes when you are in the root folder.

Reply:
I see, thanks for the explanation.